attest/verifier/types/src/convert/enclave_report_data_contents.rs

```diff
@@ -162,80 +136,42 @@ mod tests {
-        let nonce_bytes = [0x1u8; 16];
-        let key_bytes = [0x22u8; 32];
-        let custom_identity = [0x33u8; 32];
-        let report_data_1 = prost::EnclaveReportDataContents {
-            nonce: nonce_bytes.to_vec(),
-            key: key_bytes.to_vec(),
-            custom_identity: custom_identity.to_vec(),
-        };
-
-        let report_data_2 = prost::EnclaveReportDataContents {
-            nonce: nonce_bytes.to_vec(),
-            key: key_bytes.to_vec(),
-            custom_identity: custom_identity.to_vec(),
-        };
-
-        let digest_1 = report_data_1.digest32::<MerlinTranscript>(b"");
-        let digest_2 = report_data_2.digest32::<MerlinTranscript>(b"");
-        assert_eq!(digest_1, digest_2);
-
-        let mut modified_nonce = nonce_bytes.to_vec();
-        modified_nonce[0] += 1;
-        let modified_nonce_report_data = prost::EnclaveReportDataContents {
-            nonce: modified_nonce,
-            key: key_bytes.to_vec(),
-            custom_identity: custom_identity.to_vec(),
-        };
-
-        let modified_nonce_digest = modified_nonce_report_data.digest32::<MerlinTranscript>(b"");
-        assert_ne!(digest_1, modified_nonce_digest);
-
-        let mut modified_key_bytes = key_bytes.to_vec();
-        modified_key_bytes[0] += 1;
-        let modified_key_report_data = prost::EnclaveReportDataContents {
-            nonce: nonce_bytes.to_vec(),
-            key: modified_key_bytes,
-            custom_identity: custom_identity.to_vec(),
-        };
-
-        let modified_key_digest = modified_key_report_data.digest32::<MerlinTranscript>(b"");
-        assert_ne!(digest_1, modified_key_digest);
-
-        let mut modified_custom_identity = custom_identity.to_vec();
-        modified_custom_identity[0] += 1;
-        let modified_custom_identity_report_data = prost::EnclaveReportDataContents {
-            nonce: nonce_bytes.to_vec(),
-            key: key_bytes.to_vec(),
-            custom_identity: modified_custom_identity,
-        };
-
-        let modified_custom_identity_digest =
-            modified_custom_identity_report_data.digest32::<MerlinTranscript>(b"");
-        assert_ne!(digest_1, modified_custom_identity_digest);
-    }
-
-    #[test]
-    fn enclave_report_data_contents_digest_without_custom_id() {
-        let nonce_bytes = [0x2u8; 16];
-        let key_bytes = [0x33u8; 32];
-        let zeroed_custom_identity = [0x0u8; 32];
-        let report_data_without_custom_id = prost::EnclaveReportDataContents {
-            nonce: nonce_bytes.to_vec(),
-            key: key_bytes.to_vec(),
-            custom_identity: vec![],
-        };
-
-        let report_data_with_zeroed_custom_id = prost::EnclaveReportDataContents {
-            nonce: nonce_bytes.to_vec(),
-            key: key_bytes.to_vec(),
-            custom_identity: zeroed_custom_identity.to_vec(),
+        // We manually build up the digest here, to help ensure that the digest
```

Comment:
Note I removed the "without_custom_id" version as deriving Digestible defaults to Strings and Vecs being omittable.
Having an `EnclaveReportDataContents` in flight missing the key or nonce is a failure case that doesn't matter too much what the digest ends up being, so it seems unnecessary to test for presence or absence of all the fields.